### PR TITLE
[MBUILDCACHE-15] maven.build.cache.enabled parameter is broken - fix

### PR DESCRIPTION
**Pull request by maximilian-novikov-db**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryProvider.java [hunk 1]
```diff
@@ -42,8 +43,12 @@
     public RemoteCacheRepositoryProvider( CacheConfig config, PlexusContainer container )
             throws ComponentLookupException
     {
-        config.initialize();
-        String hint = config.getTransport();
+        String hint = "noop";
+        if ( config.initialize() == CacheState.INITIALIZED )
```
Comment (andrey-petrenko-db) [inline src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryProvider.java:47]: maybe implement DisabledCacheConfig or NoopCacheConfig to remove such checks across code?

**Review by andrey-petrenko-db — APPROVED** [review]

File: src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryProvider.java (hunk 1 above)
Comment (maximilian-novikov-db) [reply src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryProvider.java:47]: Agree, let's do in a next pr. This one is the hotfix.

**Comment by maximilian-novikov-db** [comment]
@gnodet could you please review/merge, it's a regression since default config was implemented

**Comment by gnodet** [comment]
@maximilian-novikov-db could you rebase on master or merge master in this branch so that we can get the tests working before merging this PR ?

**Comment by gnodet** [comment]
Fixed with https://github.com/apache/maven-build-cache-extension/pull/6

**Comment by jira-importer** [comment]
Resolve #235

**Pull request closed without merging**